async.go

```diff
@@ -160,7 +161,34 @@ func getQueryResultWithRetriesForAsyncMode(
 			logger.WithContext(ctx).Errorf("failed to decode JSON. err: %v", err)
 			return respd, err
 		}
-		if respd.Code != queryInProgressAsyncCode {
+		if respd.Code == sessionExpiredCode {
+			// Update the session token in the header and retry
+			token, _, _ := sr.TokenAccessor.GetTokens()
+			if headers[headerAuthorizationKey] != fmt.Sprintf(headerSnowflakeToken, token) {
```

Comment:
So this `if` handles the case when we use external token accessor? Maybe we should ensure, that token that we set is not empty, when custom token accessor is not set?

Reply:
Added a check for non-empty token and the related test `TestGetQueryResultTokenNotSet`.